### ISSUE TITLE
Bad recommendation for ban-types in recommended rules

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#2540]
I have the code:
```ts
public foo = (event: string, data: Object): void => {
   //do stuff
}
```
And I began using the recommended rule, but [this line says I should use `object` (lowercase) as a type instead of `Object`](https://github.com/palantir/tslint/blob/master/src/configs/recommended.ts#L33)

But `object` is not a valid type in TypeScript.

![image](https://cloud.githubusercontent.com/assets/463685/24862680/983c5ed2-1dcb-11e7-86af-541c133beb55.png)

**Comment by ajafff** [comment]
object is available in typescript 2.2

Am 10.04.2017 14:57 schrieb "Chris Barr" <notifications@github.com>:

**Comment by chrismbarr** [comment]
Hmm... yeah I see that it is after a quick search.  We are using version 2.2.1 though and that's where I'm seeing this error in Visual Studio 2017.  The project compiles just fine with no errors, but the IDE reports these errors.

Something else must be wrong... I'll close this issue and continue to search.  Thanks.

**Comment by chrismbarr** [comment]
Ah, solved.  I have VS2017 ships with the IDE tooling for TS 2.1 only
https://github.com/Microsoft/TypeScript/issues/14529

My project builds with 2.2, but the IDE is only aware of 2.1.  Lame, but now I know!
